x/mongo/driver/topology/connection.go

```diff
@@ -512,24 +513,53 @@ func (c *connection) close() error {
 }
 
 func (c *connection) closed() bool {
-	return atomic.LoadInt64(&c.state) == connDisconnected
+	if atomic.LoadInt64(&c.state) == connDisconnected {
+		return true
+	}
+
+	// If the connection has been idle for less than 10 seconds, skip the liveness
+	// check.
+	idleStart, ok := c.idleStart.Load().(time.Time)
+	if !ok || idleStart.Add(10*time.Second).After(time.Now()) {
```

Comment:
My reference was the PyMongo driver, which [checks connection liveness when it's idle for >1 second](https://github.com/mongodb/mongo-python-driver/blob/e6b95f65953e01e435004af069a6976473eaf841/pymongo/synchronous/pool.py#L983-L985). I wanted to mitigate the risk that checking liveness too often would cause performance problems (since a liveness check takes at least 1ms), so I increased that threshold to 10s.